mlflow/gateway/providers/openai.py

```diff
@@ -326,10 +530,10 @@ async def _chat_uc_function(self, payload: chat.RequestPayload) -> chat.Response
         else:
             # No UC functions to execute
             resp = await send_request(
-                headers=self._request_headers,
-                base_url=self._request_base_url,
+                headers=self.headers,
+                base_url=self.base_url,
                 path="chat/completions",
-                payload=self._add_model_to_payload_if_necessary(payload),
+                payload=OpenAIAdapter.chat_to_model(payload, self.config),
```

Comment:
We can `payload=self.adapter_class.chat_to_model(payload, self.config)`, 
and seemingly we can move the block code to base provider class.
There are several other similar cases. We can do some refactoring if possible. (This comment does not block this PR).

Reply:
> We can use payload=self.adapter_class.chat_to_model(payload, self.config)

Great catch!

> and seemingly we can move the block code to base provider class.

I agree we can do some refactoring, but not sure how easy it is because most of the implementations are done by community and may not be maintained well.